sdk/tables/test-resources.json

```diff
@@ -0,0 +1,365 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "baseName": {
+      "type": "string",
+      "metadata": {
+        "description": "The base resource name."
+      }
+    },
+    "testApplicationOid": {
+      "type": "string",
+      "metadata": {
+        "description": "The client OID to grant access to test resources."
+      }
+    },
+    "tenantId": {
+      "type": "string",
+      "defaultValue": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+      "metadata": {
+        "description": "The tenant ID to which the application and resources belong."
+      }
+    },
+    "testApplicationId": {
+      "type": "string",
+      "metadata": {
+        "description": "The application client ID used to run tests."
+      }
+    },
+    "testApplicationSecret": {
+      "type": "string",
+      "metadata": {
+        "description": "The application client secret used to run tests."
+      }
+    },
+    "enableVersioning": {
+      "type": "bool",
+      "defaultValue": true
+    },
+    "storageEndpointSuffix": {
+      "type": "string",
+      "defaultValue": "core.windows.net",
+      "metadata": {
+        "description": "Storage endpoint suffix. The default value uses Azure Public Cloud (core.windows.net)"
+      }
+    }
+  },
+  "variables": {
+    "storageApiVersion": "2022-05-01",
+    "location": "[resourceGroup().location]",
+    "accountName": "[parameters('baseName')]",
+    "datalakeAccountName": "[concat('d', parameters('baseName'))]",
+    "accountNameTidy": "[toLower(trim(variables('accountName')))]",
+    "datalakeaccountNameTidy": "[toLower(trim(variables('datalakeAccountName')))]",
+    "blobEndPoint": "[concat('https://',variables('accountNameTidy'),'.blob.', parameters('storageEndpointSuffix'))]",
+    "accountSasProperties": {
+      "signedServices": "bfqt",
+      "signedPermission": "rwdlacup",
+      "signedResourceTypes": "sco",
+      "keyToSign": "key1",
+      "signedExpiry": "2099-01-01T23:59:00Z"
+    },
+    "authorizationApiVersion": "2018-01-01-preview",
+    "blobDataContributorRoleId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/ba92f5b4-2d11-453d-a403-e96b0029c9fe')]",
```

Comment:
i'm not touching that with a ten foot pole, it works and that's all i care about, this is a very finicky json that i barely got working , these templates are a pain in the neck to edit/create

Reply:
I understand it's likely complicated and prone to breaking. Let's revisit this separately in an issue to update such infra scripts across the repo, since relying on preview service and API versions is not a good practice (as far as I am aware), especially ones that are 4+ years old (even if they are currently "working").